emcc.py

```diff
@@ -2058,6 +2068,15 @@ def do_minify(): # minifies the code. this is also when we do certain optimizati
           for script in shared.Settings.BINARYEN_SCRIPTS.split(','):
             logging.debug('running binaryen script: ' + script)
             subprocess.check_call([shared.PYTHON, os.path.join(binaryen_scripts, script), js_target, wasm_text_target], env=script_env)
+        # after generating the wasm, do some final operations
+        if not shared.Settings.WASM_BACKEND:
+          if shared.Settings.SIDE_MODULE:
+            wso = shared.WebAssembly.make_shared_library(js_target, wasm_binary_target)
+            # replace the .js output with the shared library. TODO: emit a file with suffix .wso
```

Comment:
Yeah, maybe another name is better. I'll edit the TODO to not mention a specific one, and use "dynamic library" as the term.

`.dylib` sounds a little too OSX-ey to me. But maybe `wylib`, `dywasm`, or some other form...